--- src/library/scala/runtime/ScalaRunTime.scala	2018-03-22 12:22:03.000000000 +0100
+++ test/instrumented/library/scala/runtime/ScalaRunTime.scala	2018-04-05 22:52:10.000000000 +0200
@@ -10,6 +10,8 @@
  * additional information regarding copyright ownership.
  */
 
+/* INSTRUMENTED VERSION */
+
 package scala
 package runtime
 
@@ -52,8 +54,11 @@
   def anyValClass[T <: AnyVal : ClassTag](value: T): jClass[T] =
     classTag[T].runtimeClass.asInstanceOf[jClass[T]]
 
+  var arrayApplyCount = 0
+
   /** Retrieve generic array element */
   def array_apply(xs: AnyRef, idx: Int): Any = {
+    arrayApplyCount += 1
     xs match {
       case x: Array[AnyRef]  => x(idx).asInstanceOf[Any]
       case x: Array[Int]     => x(idx).asInstanceOf[Any]
@@ -69,8 +74,11 @@
     }
   }
 
+  var arrayUpdateCount = 0
+
   /** update generic array element */
   def array_update(xs: AnyRef, idx: Int, value: Any): Unit = {
+    arrayUpdateCount += 1
     xs match {
       case x: Array[AnyRef]  => x(idx) = value.asInstanceOf[AnyRef]
       case x: Array[Int]     => x(idx) = value.asInstanceOf[Int]
